src/coreclr/jit/valuenum.cpp

```diff
@@ -8005,24 +8008,30 @@ void Compiler::fgValueNumberTree(GenTree* tree)
                                             vnStore->VNPairApplySelectorsAssign(oldLhsVNPair, fieldSeq, rhsVNPair,
                                                                                 lhs->TypeGet(), compCurBB);
                                     }
-                                    lvaTable[lclNum].GetPerSsaData(lclDefSsaNum)->m_vnPair = newLhsVNPair;
+
+                                    if (lclDefSsaNum != SsaConfig::RESERVED_SSA_NUM)
+                                    {
+                                        lvaTable[lclNum].GetPerSsaData(lclDefSsaNum)->m_vnPair = newLhsVNPair;
+                                        wasInSsa                                               = true;
```

Comment:
This is also a fix here as well.

Reply:
Right, this is another place where we could potentially call `GetPerSsaData` with a bad SSA num. But here it was a bit harder to get the control flow the same as before since the tree is not available when we call `lvaInSsa` above, so I had to introduce the variable.